diff --git a/dist/index.js b/dist/index.js
index 8070b46bb86bfdc5212c18a2c5f7ff9a389411dd..4b518a75717c3b0baed274c7bfef3db4e8340150 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -61,14 +61,11 @@ class EIP1193 extends types_1.Connector {
     activate() {
         return __awaiter(this, void 0, void 0, function* () {
             this.actions.startActivation();
-            return Promise.all([
-                this.provider.request({ method: 'eth_chainId' }),
-                this.provider
-                    .request({ method: 'eth_requestAccounts' })
-                    .catch(() => this.provider.request({ method: 'eth_accounts' })),
-            ])
-                .then(([chainId, accounts]) => {
-                this.actions.update({ chainId: parseChainId(chainId), accounts });
+            return this.provider.request({ method: 'eth_chainId' }).then((chainId) => {
+                this.actions.update({ chainId: parseChainId(chainId) })
+                return this.provider.request({ method: 'eth_requestAccounts' })
+            }).then((accounts) => {
+                this.actions.update({ accounts });  
             })
                 .catch((error) => {
                 this.actions.reportError(error);
